File: src/kubeview/api/exec.js
import { invoke } from "@tauri-apps/api/core";
import { listen } from "@tauri-apps/api/event";

export function execPodShell(context, namespace, pod, container) {
  return invoke("exec_pod_shell", {
    context: context ?? null, namespace, pod, container: container ?? null,
  });
}

export function execPodStdin(sessionId, data) {
  return invoke("exec_pod_stdin", { sessionId, data });
}

export function execPodStop(sessionId) {
  return invoke("exec_pod_stop", { sessionId });
}

export async function onShellOutput(callback) {
  return listen("shell-output", (event) => {
    callback(event.payload);
  });
}

File: src-tauri/src/commands/exec.rs
use crate::kube;

#[tauri::command]
pub(crate) async fn exec_pod_shell(
    app: tauri::AppHandle,
    context: Option<String>,
    namespace: String,
    pod: String,
    container: Option<String>,
) -> Result<String, String> {
    kube::exec_pod_shell(context, namespace, pod, container, app).await
}

#[tauri::command]
pub(crate) async fn exec_pod_stdin(
    session_id: String,
    data: String,
) -> Result<(), String> {
    kube::exec_pod_stdin(session_id, data).await
}

#[tauri::command]
pub(crate) async fn exec_pod_stop(
    session_id: String,
) -> Result<(), String> {
    kube::exec_pod_stop(session_id).await
}
